store/store.go

```diff
@@ -458,6 +415,8 @@ func (s *Store[H]) Append(ctx context.Context, headers ...H) error {
 	// queue headers to be written on disk
 	select {
 	case s.writes <- headers:
+		// we return an error here after writing,
+		// as there might be an invalid header in between of a given range
```

Comment:
I don't recall I added this. Was it you @Wondertan ?

Reply:
https://github.com/celestiaorg/go-header/pull/274#discussion_r2042031151

yes, rebase artifact

Reply:
Removed.